### ISSUE TITLE
German localization - latest addenda

### ISSUE THREAD

**Issue by papioara** [frescobaldi/frescobaldi#1669]
Today I have taken care of the latest version of the de.po file in frescobaldi/frescobaldi for translation, i.e. the newly added strings, and almost invariably there have come up some questions:

1. The most cryptic one first: what is the context of 'Part {self.__class__.__name__}'? Where is it to be found?
2. 'Custom...' - context again...
3. 'Syn.Fx.' - it had been translated 'Syn.Ld.', but I am afraid this is not correct. I'd rather leave it as it is in English, because in the world of artificial sounds so much is in English anyway. What is your opinion?
4. There are two entries about embedding LilyPond source code in a PDF, where there is an addition in the German translation - not by me - about this being possible since version 2.19.39. Is this information correct? If so, it can remain there, although the English original does not mention it.
5. Abbreviation for Dulcimer: I have consulted quite a few sources, but I could not find any official abbreviation for the German word 'Hackbrett', since this instrument does never appear in orchestral scores, where they are needed and used mostly.

So much for the new stuff. I'll proof-read all the rest, too, so no unnoticed change can fall through, but I hope that there won't be any more findings.

**Comment by papioara** [comment]
Hi,
back again after a few days off in the garden...
Thanks for your comments as usual: now I know my way around items 1. - 4.
Question 5 cannot have a meaningful answer at the moment, so I'll just leave my made-up 'official' translation in there. At least every German speaker can tell from it what it means, and you save 2(!) characters  ;-)

To answer your last remark: at the beginning, seeing some wrong translations in poedit, I was a little wary about the non-fuzzy strings. I have decided to go over all of them. It was a good decision, because I have discovered (very few, but still...) some small typos, and I have changed (for the better, I hope) some strings (also very few). It's good to see this again after a while with a little
'recul' - I do not know a good English word, so I'm using a French one... 
It'll take longer, and I still have to work my way through everything, but then I will really be sure I have done a sufficiently decent job. Localising Frescobaldi is not an easy job, because the matter is complex. It's technically powerful and packed with all kinds of features, and on top of that you have the things related to music and notation.

**Comment by papioara** [comment]
It's a lot more work than I thought, because I still needed to do a lot of fine-tuning. Seems that I did not give enough attention to all the strings already translated. Still not done completely. Have learned to detect contexts (where to find strings in Frescobaldi in general) from the comments above the 'msgid' lines - very helpful indeed! We'll have a more polished localization afterwards. PR coming soon, I hope.

**Comment by papioara** [comment]
I've done it! Went through **all** of the 2006 strings and really looking at **everything**, doing research if necessary for clarification, searching in Frescobaldi, the comments in the de.po file on Github etc. and was very successful about finding answers to my queries. The result is a lot of small corrections, precisions, better re-wordings - in other words: a thoroughly revised and enhanced version.

Before handing it in, though, it's question time again, because a couple of things were impossible for me to find out about. Here is the list:

1. Syn.Fx: The former version (not by me) was Syn.Ld. Guess this is not correct, right? It should remain Syn.Fx
2. Alt: Is this about the ALT key? The messagectxt points in that direction, but I want to make sure...
3. Clear Grab: What does this do? What is a grab in this context? What is being grabbed?
4. Top Menu: Also seems to be in the shortcut context, but what does it do? Is it about the Main Menu Bar at the top of the window?
5. SysReg: Another one having to do with shortcuts. But what does it mean?
6. Attribute type of wrapper: s this about an attribute type to be assigned to the wrapper, or is it an attribute type for/of a wrapper? German-specific semantic difficulty...
7. Rotate print on regular paper: What does this mean exactly? Cannot find it in some menu. Could it be: Print landscape on paper with the usual portrait orientation when it sits in the printer? Wording sounds confusing to me.
8. ENC: What is this? What does it stand for?
9. Failed extensions: What exactly does this mean? Extensions which could not be loaded? Extensions which did not work as expected?
10. Failed dependencies: Similar to 9. Missing dependencies? Faulty dependencies (wrong version e.g.)?
11. Complex score, requires a 'clean' font: Are 'clean' fonts a font family? Never heard of them before... Please confirm...
12. Select Music View Document: Where in Frescobaldi can I find it? Is it the corresponding function like in the Manuscript Viewer (Select Manuscript Document)? If so: I cannot see a toolbar widget in the Music Viewer, but well in the Manuscript Viewer.

That's it. Hope for clarification. :-)

**Comment by papioara** [comment]
Thanks for your answers - very use- and helpful. 1, 3 and 4 are still missing, but I hope they'll come a little later ;-)
As for your question about 5. SysReg: it's my bad and a tpyo. I mean 'SysReq' (see screenshot)

![Auswahl_070](https://github.com/frescobaldi/frescobaldi/assets/31970041/b4d5e4fb-f382-4526-aad8-bf8b22cbd7e5)

I am referring to this here in question 11:

![Auswahl_071](https://github.com/frescobaldi/frescobaldi/assets/31970041/ec053371-401a-404f-9690-e5dae19c7c28)

It's very well possible what you say. In that case 'clean' just would be a description to characterize the kind of font needed in such kind of a score. But I cannot 'conjure up' that tooltip you mention in Tools > Document Fonts - see screenshot. Where does it hide?

![Auswahl_072](https://github.com/frescobaldi/frescobaldi/assets/31970041/5bca7c5a-a9f7-48e5-a5ee-ebae4c2db92b)

**Comment by papioara** [comment]
Strange: when I do so, nothing appears, whereas in other places in the same window, tooltips pop up, for instance over 'Complete \paper block'.

**Comment by papioara** [comment]
Actually none of the tooltips accompanying the various choices there like 'Bach', 'Scriabine' etc. shows in my local Frescobaldi installation - don't know what's wrong, but for the moment do not take it too seriously...

**Comment by ksnortum** [comment]
Slightly off-topic, but shouldn't it be Scriabin, not Scriabine?

**Comment by papioara** [comment]
Yes, according to Wikipedia and to the Grove's Dictionary of Music. During my localization work I also have discovered some minor things which could need some improvement, e.g. typos like 'unckecked', or at least disputable punctuation issues and the like. But as a German speaker I rather leave this to the native English speakers. 2006 text strings to sift through is already a lot for one language...

**Comment by papioara** [comment]
@jeanas : You are right: you have to click and then hover over the pop-up menu item. It's counter-intuitive, because most other tooltips appear when you hover over some object in a window. Never mind: I translated the 'clean' font according to your proposal. I think it really makes sense. Thank you.

**Comment by papioara** [comment]
Another question just came up: what does 'Check' mean in the context of Preferences > Fonts & Colors > Lilypond? Does it have something to do with 'bar check' or 'octave check'? You can assign a color to it, but by default it seems to be left blank.

**Comment by papioara** [comment]
New question: There's a snippet 'Title case selection'. How can you apply it? Selecting it in the Snippet Manager and selecting 'Apply' does not do anything at all. I assume it lets you change the case in the title - all words at once. Right or something else?

**Comment by papioara** [comment]
Ah, that's how it works! Do you write every word in a title with capital letters in English? Even if it is not a noun? If so, this is a very special snippet and would not be useful in German, for instance. And it seems you cannot toggle. If you are unhappy with the result, you have to undo manually. Oh well, it's also just one keystroke...

**Comment by papioara** [comment]
And what does this here mean or do:

`#: ../../i18n/messages.py:176`
`msgctxt "QShortcut"`
`msgid "Application Right"`
`msgstr "Anwendung rechts"`

There's also 'Application Left'. What kind of a message is it?

**Comment by papioara** [comment]
OK, so I'll just leave it alone. Translated it, but neither the English nor the German string make sense to me. It's like with 'Away'...

**Comment by papioara** [comment]
And maybe this one:

`msgctxt "QShortcut"`
`msgid "Clear Grab"`
`msgstr "Grab(?) löschen"`

Any idea about what has been grabbed from where and needs to be disposed of?

**Comment by papioara** [comment]
And this one:

`msgctxt "QShortcut"`
`msgid "SysReq"`
`msgstr "SysReq"`

System Requirements??

**Comment by papioara** [comment]
And lastly:

`#: ../../i18n/messages.py:269`
`msgctxt "QShortcut"`
`msgid "Top Menu"`
`msgstr "Hauptmenü"`

Which menu is this talking about?

And our old uninvited friend still seems to be here:

`'Part {self.__class__.__name__}'`

That's it for today. I've really been productive on de.po - made it to the end at last (one more time).

**Comment by papioara** [comment]
Did a test drive in my local git branch, and everything so far looks excellent. There are those open questions from my last 3 comments plus two other things which I noticed:

- The stubborn 'Close' button in the score preview in the Score Wizard and in Tools > Document Fonts. The former has also been confirmed for the Italian localization.
- A stubborn menu item (translation exists in de.po - I double-checked) in Documentation Browser. In the panel toolbar there is a globe widget, and when you opt for submenus on widgets in the settings (the little triangle next to the widget which will show up then), you get to see one option (Open Current Page in Web Browser) in English, and the second one as expected in German.

Would that be easy to explain or would it mean elaborate research into possible causes?
And: what about fixing the little flaws in the English strings? It's mostly minor stuff, of course, but there are also some typos like 'unckecked' for instance. Would that be more complex than doing a localization? Just asking...

**Comment by papioara** [comment]
I have sifted through de.po and found the following items saying 'Close' exclusively (just that one word) in some or the other form:
No.1
`#: ../../i18n/messages.py:197`
`msgctxt "QShortcut"`
`msgid "Close"`
`msgstr "Schließen"`

No.2
#: ../../i18n/messages.py:155
#, fuzzy
`msgctxt "QPlatformTheme"`
`msgid "Close"`
`msgstr "Schlie&ßen"`

No.3
`#: ../../i18n/messages.py:138`
`msgctxt "QGnomeTheme"`
`msgid "&Close"`
`msgstr "Schlie&ßen"`

No.4
`#: ../../i18n/messages.py:89`
`msgctxt "QDockWidget"`
`msgid "Close"`
`msgstr "Schlie&ßen"`

No.5
`#: ../../i18n/messages.py:74`
`msgctxt "QDialogButtonBox"`
`msgid "Close"`
`msgstr "Schließen"`


No.6
`#: ../../frescobaldi_app/viewers/contextmenu.py:129`
`msgid "Close..."`
`msgstr "Schließen..."`

Would that give someone a clue who knows more than me?

**Comment by papioara** [comment]
Thanks for asking. I have not been able to track down an untranslated 'Close' string, but I have edited the de.po file again and unfuzzied a couple of other fuzzy items which actually have been clarified in the meantime. Plus I removed every ampersand from the 'Close' translations except for one case, where it also was in the English original and, of course, did `tox -e mo-generate.`.
And lo and behold: everything is right now - the stubborn buttons have given in, and the menu item in the Documentation Browser is in German as well.
That would inciate that at least in the Italian localization these issues could also be fixed, I guess. They are still there, when you switch to Italian.

So, apart from the couple of still open questions (sorry for being so insisting ;-) ), the localization seems to be complete. As far as I can see, there are no English leftovers, and it really looks polished and like professional software, which it is anyway. 

As someone who is absolutely in favour of the Open Source ideals, I know about the limitations and problems which can come with it. We all, users and developers alike, depend on people who are willing to give their time and talent to contribute to a project. Many projects just do not have the resources or do not attract enough contributors who help with secondary things such as localizations. But they are important enough, because they can help tapping into a greater user base and by making an application easier to use.

Just want to give something back - have been using Frescobaldi such a lot for many years...

**Comment by papioara** [comment]
Thank you for your incessant and indefatigable support! I've never heard of such a key before - very interesting, and probably I'm in good company.
Thanks for fixing the typos. Maybe there are more of them (but very rarely, thank God), but these were the ones I remembered.
But there are also strings in English which raise questions with me, because either they are there in the present form intentionally, or because it's just some form of nonchalance:

1. is there a reason why some strings do not begin with a capital letter? Are there any rules which determine if a menu item has to be all capital letters for each individual word or only the first word with a capital initial letter? (Examples: Documentation Browser, or Save documents in this folder, and lastly all lowercase: bottom of last page)
2. Punctuation:
- 1.   I have observed that many strings, although they are not a sentence, have a full stop at the end. This looks strange and strictly speaking: it is unnecessary. But maybe in the English-speaking world this is different? If so: there are also strings of the same kind which do not have one. Wouldn't it be a good idea to make these things consistent? Or there is a full stop, but there should be a colon. Or there should be a full stop, but there is nothing. Or there's a colon, but there should be nothing rather. It's rather mixed, as you can see. Nevertheless the overall number of changes will not be deterring. But opting for change: yes or no would also have to take into account the question of how much impact the changes would have on existing translations, if you say that there are not many active translators. What's your opinion on this?
- 2.   Is there a rule about when a menu item/command should be followed by '...'?

**Comment by papioara** [comment]
> It's not hard, but a little tedious.
This is all tedious work - like proof-reading for a publisher. But I'd say it's worth it, because afterwards the outer appearance will be more flawless. In an age where outer appearance is paramount, this is an advantage ;-)

**Comment by papioara** [comment]
> For example, for names with a specific capitalization like "openLilyLib", or in the --help output.

I agree with that, of course. This is like proper names: they need to always remain the same. I was rather thinking of plain normal English sentences/expressions like the one I already mentioned: 'bottom of last page'. Maybe there are different conventions/guidelines for either menu items, tool-tips or strings which appear inside a window like, say, the Score Settings one in the Score Wizard?

I did not know about the ellipsis. Good to see that this is not random...

So after all the amount of candidates for improvement has become smaller already. Would it be a good idea to create a spreadsheet containing everything I think should be changed plus a proposal for corrections? Reading through all the strings in English only is a lot faster than translating them. What do you think? That might provide a more solid foundation for further proceedings.

**Comment by papioara** [comment]
Yes, I could do all this in principle. I do not think that this is witchcraft and can be learned - I never have done something like this before, but I am willing to learn. After all, this is for a good purpose.

I will create my spreadsheet, because it will give me a better idea of the quantity of changes necessary in my opinion. And maybe I should at least show it to you or anyone else for approval. I also have a good friend who is a native speaker, and who does translations himself, whom I could ask for supervision. Like that, I would be on the safe side as far as grammar and punctuation are concerned.

When this is done, I can go about the technical stuff. Basically, I have two questions about it: 

1. Where in Frescobaldi do I find the source code with the English strings? I probably will have to read something about correct formatting before changing anything, I guess.
2. We have 18 .po files which would need updating afterwards. This probably has to be done manually, but I expect that this (depending on the amount of changes) will not take more than a day or so. I have never delved very deeply into scripting, so here we are. I will to make all these changes including the ones in the source, in my local branch. And afterwards: should I submit one big PR containing _all_ the changes or individual PRs rather? What is easier to handle for you?

**Comment by papioara** [comment]
Thank you, now I understand even better what you mean by tedious, but I'll stand by my offer. I am already making a list of potential changes. So far it's actually mainly punctuation stuff.

**Comment by papioara** [comment]
You mean that the changed entries will be marked fuzzy?

**Comment by papioara** [comment]
I am half way through the English strings and will stop for today. Found 78 candidates in 5.100 out of 10.000 lines...

**Comment by papioara** [comment]
Ah, that's what it is! This is really obscure! Maybe I'll come up with a clever translation nevertheless, but if not: you are right that this is a very improbable scenario.
Thanks a lot for your effort in any case. I just want to be really complete, and then even such stuff needs to be explained. You never know if maybe it could be important.

**Comment by papioara** [comment]
Just a small in-between report: I am in the middle of unfuzzying strings in the po files - nine down, nine still to go. The English strings are modified, but not very many (46 in all). I have run `tox -e po-update` for every single language.
Everything seems to have worked without problems so far. When unfuzzying strings, I also have corrected some all-too-obvious mistakes like missing '{name}' and stuff poedit was complaining about. Otherwise I have unfuzzied everything globally, so everything will show up in the Frescobaldi menus/tool-tips/messages/Preferences and whatever. Degrees of fuzziness (or is it fuzzyness??) vary a lot to my surprise. And also completeness. But of course this is Open Source and localisation: you can only work with what comes in from contributors.

In a couple of days I should be done with everything and be ready for the PR. Maybe I might have the one or other technical question beforehand - we'll see.

**Comment by papioara** [comment]
> What do you mean by "I have unfuzzied everything globally"?

Your question sounds like I messed up...
To be more precise: I have removed **_all_** the 'fuzzy' flags so far in nine po files: cs, de, es, fr, gl, it, ja, ko, and nl. It seems I may have got you wrong here. Should I have looked **_only_** for the 'fuzzy' entries corresponding to my changes in the English originals for removal of the flag?

If this is what you actually meant: is there a simple way to undo these changes? In most languages, the number of fuzzy entries is not that large thank God. 

Galician was by far the worst, because it is pretty fragmentary anyway and was really full of fuzzy strings which definitely needed revision for various reasons. This was obvious even to somebody who does not speak Galician, but understands some Spanish, French, or Italian.

**Comment by papioara** [comment]
Thank you very much! You are an excellent pilot with whose help I hopefully can make my boat (which I have run aground) afloat again. ;-)

I have one other question: when I use poedit, make changes and save them afterwards, poedit creates/updates the corresponding .mo file. Is this technically the same as doing `tox -e mo-generate`? Or does poedit only update the .mo file of the LANG.po I have been working on, and tox generates .mo files for all of the LANG.po's?

**Comment by papioara** [comment]
Phew, I got myself out of the mess...
I had not committed anything yet, and so restoring was a one-step procedure only. A spot check of a couple of .po's showed me that the fuzzy strings were back again.

Because this is the easiest one for me, I scrutinized de.po and found that the fuzzy strings corresponded to my changes in the English text strings indeed. I have unfuzzied them one by one according to my list (I'm really glad I have it now). And now I will have to do this manually seventeen more times for the remaining languages, because as far as scripting is concerned, I am not very knowledgable. I never got beyond quite simple ones, because I have never needed it.

Be it as it is: I will survive this - it only will take some time. Hopefully I can do one or two languages a day, so it won't take too long.

**Comment by papioara** [comment]
I did it! All 18 languages are unfuzzied in the right places now. To be on the safe side, I saved every single LANG.po in poedit, and after they all had been processed, i.e. edited, I ran `tox -e mo-generate` as well.

There were some small surprises, though, in the process. There were entries I expected to be fuzzy after my changes, but they weren't - every now and then - mostly they met my expectations. And there were also some English strings I thought I had changed (moved a colon, deleted a full stop, replaced a colon by a full stop, adding a space and such things mostly) which weren't different from the original version.
Because it was also not very frequent, I left these strings as they were. After all: it's small details.

But I also had to make a couple of entries fuzzy, although they weren't marked as such. Some translations obviously were incorrect. Sometimes you can even see this in Chinese, or at least conclude. When I threw those strings at DeepL (a German AI translator), my suspicions were confirmed. So now (not only in Chinese, but also in Ukrainian, Russian e.g. (which I can read at least) there are some new fuzzy strings now.

Hope you agree with my decisions...

I did not commit anything yet in case you have other ideas. I have documented everything in my spreadsheet, so it won't be difficult to change things if necessary.

**Comment by papioara** [comment]
> Hm, are you sure you ran tox -e po-update -- <LANG> for these languages?

Yes, I am. I double-checked with the i18n/frescobaldi folder on Github. And there were only a couple which had remained unchanged. And it were always the same ones in all the languages if I remember correctly.

I just have looked at my spreadsheet log again, and statistically speaking, by far most of the entries were ok. As an example for what I left fuzzy (although my change of the English string was there) were missing `{name}` or `{url}` entries in Chinese for instance which by poedit were mentioned as errors after the check. I have no way of knowing where to insert that, and so I left them as they were. 
And one more thing I have to say about the translations in general (probably this is nothing new for you): their quality is going from quite sketchy and probably having been abandoned after an initial effort up to being complete. The former ones have lots of fuzzy entries - sometimes hundreds - and the complete ones practically only had those which by my changes had become fuzzy.

Anyway - I'm going to get ready for the PR. You'll have it tonight hopefully, so you can take a look as well.
